arangod/RocksDBEngine/RocksDBRestReplicationHandler.cpp

```diff
@@ -888,13 +888,24 @@ void RocksDBRestReplicationHandler::handleCommandRevisionTree() {
   // smaller and thus improve efficiency)
   bool onlyPopulated = _request->parsedValue("onlyPopulated", false);
 
-  auto tree = ctx.collection->getPhysical()->revisionTree(ctx.batchId);
+  std::string collectionName = _request->value("collection");
+
+  std::unique_ptr<containers::RevisionTree> tree;
 
   {
     RocksDBReplicationContext* c = _manager->find(ctx.batchId);
     RocksDBReplicationContextGuard guard(_manager, c);
+    if (c != nullptr) {  // we have the RocksDBReplicationContext!
+      // See if we can get the revision tree from the context:
+      tree = c->getPrefetchedRevisionTree(collectionName);
+      // This might return a nullptr!
+    }
+    if (tree == nullptr) {  // Still not there, try to get it directly:
+      tree = ctx.collection->getPhysical()->revisionTree(
+          c->snapshot()->GetSequenceNumber());
```

Comment:
In theory, `snapshot()` can return a nullptr. While it is very unlikely that it does here, it is still possible. So we should check here if the returned snapshot is a nullptr or not.

Reply:
Good point, added a check,will fail if there is no snapshot.